src/modules/users/users.controller.ts

```diff
@@ -1,31 +1,83 @@
-import { Body, Controller, Get, Post } from '@nestjs/common';
+import { Body, Controller, Get, Post, Delete, Param } from '@nestjs/common';
 import { UsersService } from './users.service';
-import { User } from 'src/common/entities/user.entity';
-import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { User } from './user.entity';
+import {
+  ApiOperation,
+  ApiOkResponse,
+  ApiInternalServerErrorResponse,
+  ApiResponse,
+  ApiTags,
+  ApiBody,
+  ApiParam,
+} from '@nestjs/swagger';
+import { CreateUserDTO } from './DTO/create.dto';
+import { ResponseUserDTO } from './DTO/response.dto';
+import { ErrorResponse } from 'src/common/ErrorResponse';
+import { responseDescrptions } from 'src/common/responseDescriptions';
 
 @ApiTags('users')
```

Comment:
I assume you need to have this route protected as now anyone can make the request (login and not login user)